src/library/scala/collection/Iterator.scala

```diff
@@ -83,6 +83,9 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
     */
   def hasNext: Boolean
 
+  @deprecated("hasDefiniteSize on Iterator is the same as isEmpty", "2.13.0")
+  @`inline` override final def hasDefiniteSize = isEmpty
```

Comment:
Is `hasDefiniteSize` supposed to perform computation? because `isEmpty` can potentially force the iterator to evaluate elements. Would `knownSize <= 0` be a better implementation?

Reply:
`knownSize >=0` would always be better but this is what 2.12 does